### ISSUE TITLE
Crash NotSerializableException io.getstream.chat.android.ui.message.list.MessageListItemStyle

### ISSUE THREAD

**Issue by spvince** [GetStream/stream-chat-android#2688]
**Describe the bug**
Crash in my firebase crashlytics.

**SDK version**
- 4.22.0

**Crash**
```
Fatal Exception: java.lang.RuntimeException: Parcelable encountered IOException writing serializable object (name = io.getstream.chat.android.ui.message.list.background.MessageBackgroundFactoryImpl)
       at android.os.Parcel.writeSerializable(Parcel.java:1836)
       at android.os.Parcel.writeValue(Parcel.java:1783)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at androidx.fragment.app.FragmentState.writeToParcel(FragmentState.java:159)
       at android.os.Parcel.writeTypedObject(Parcel.java:1637)
       at android.os.Parcel.writeTypedList(Parcel.java:1516)
       at android.os.Parcel.writeTypedList(Parcel.java:1473)
       at androidx.fragment.app.FragmentManagerState.writeToParcel(FragmentManagerState.java:64)
       at android.os.Parcel.writeParcelable(Parcel.java:1804)
       at android.os.Parcel.writeValue(Parcel.java:1710)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at android.os.Parcel.writeValue(Parcel.java:1701)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at android.os.Parcel.writeValue(Parcel.java:1701)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.app.IActivityTaskManager$Stub$Proxy.activityStopped(IActivityTaskManager.java:4627)
       at android.app.servertransaction.PendingTransactionActions$StopInfo.run(PendingTransactionActions.java:145)
       at android.os.Handler.handleCallback(Handler.java:914)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:225)
       at android.app.ActivityThread.main(ActivityThread.java:7564)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

```
Caused by java.io.NotSerializableException: io.getstream.chat.android.ui.message.list.MessageListItemStyle
       at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1240)
       at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1604)
       at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1565)
       at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1488)
       at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1234)
       at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
       at android.os.Parcel.writeSerializable(Parcel.java:1831)
       at android.os.Parcel.writeValue(Parcel.java:1783)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at androidx.fragment.app.FragmentState.writeToParcel(FragmentState.java:159)
       at android.os.Parcel.writeTypedObject(Parcel.java:1637)
       at android.os.Parcel.writeTypedList(Parcel.java:1516)
       at android.os.Parcel.writeTypedList(Parcel.java:1473)
       at androidx.fragment.app.FragmentManagerState.writeToParcel(FragmentManagerState.java:64)
       at android.os.Parcel.writeParcelable(Parcel.java:1804)
       at android.os.Parcel.writeValue(Parcel.java:1710)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at android.os.Parcel.writeValue(Parcel.java:1701)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.os.Parcel.writeBundle(Parcel.java:1000)
       at android.os.Parcel.writeValue(Parcel.java:1701)
       at android.os.Parcel.writeArrayMapInternal(Parcel.java:931)
       at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1593)
       at android.os.Bundle.writeToParcel(Bundle.java:1253)
       at android.app.IActivityTaskManager$Stub$Proxy.activityStopped(IActivityTaskManager.java:4627)
       at android.app.servertransaction.PendingTransactionActions$StopInfo.run(PendingTransactionActions.java:145)
       at android.os.Handler.handleCallback(Handler.java:914)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:225)
       at android.app.ActivityThread.main(ActivityThread.java:7564)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

**Device:**
 - Vendor and model: Redmi Note 8
 - Android version: 10

**Comment by ogkuzmin** [comment]
Thanks for submitting the issue! We will investigate it

**Comment by ogkuzmin** [comment]
We found the root cause and the fix is on our review. It will be available in the new version (should be available in two weeks)

**Comment by adasiewiczr** [comment]
hey @spvince,
the fix is available in the latest release (4.24.0). Check our [CHANGELOG](https://github.com/GetStream/stream-chat-android/blob/main/CHANGELOG.md#december-9th-2021---4240) for further details
